manager/controlapi/task.go

```diff
@@ -104,10 +106,10 @@ func (s *Server) ListTasks(ctx context.Context, request *api.ListTasksRequest) (
 	if request.Filters != nil {
 		tasks = filterTasks(tasks,
 			func(e *api.Task) bool {
-				return filterContains(e.ServiceAnnotations.Name, request.Filters.Names)
+				return filterContains(e.ServiceAnnotations.Name+"."+strconv.FormatUint(e.Slot, 10), request.Filters.Names)
```

Comment:
Thanks @stevvooe the pull request has been updated with `fmt.Sprintf`. Please let me know if there are any issues.